src/main/webapp/app/course/manage/course-detail.component.ts

```diff
@@ -62,16 +128,52 @@ export class CourseDetailComponent implements OnInit, OnDestroy {
      */
     ngOnDestroy() {
         this.eventManager.destroy(this.eventSubscriber);
+        this.dialogErrorSource.unsubscribe();
+        this.websocketService.unsubscribe(CourseDetailComponent.getCourseArchiveStateTopic(this.course.id!));
     }
 
-    /**
-     * Subscribe to changes in courses and reload the course after a change.
-     */
-    registerChangeInCourses() {
-        this.eventSubscriber = this.eventManager.subscribe('courseListModification', () => {
-            this.courseService.find(this.course.id!).subscribe((courseResponse: HttpResponse<Course>) => {
-                this.course = courseResponse.body!;
-            });
-        });
+    private static getCourseArchiveStateTopic(courseId: number) {
+        return '/topic/courses/' + courseId + '/archive-course';
+    }
+
+    canArchiveCourse() {
+        // A course can only be archived if it's over.
+        const isCourseOver = this.course.endDate?.isBefore(moment()) ?? false;
```

Comment:
Mini improvement, I think this can be shortened to `const isCourseOver = this.course.endDate?.isBefore(moment());`

Reply:
It technically can but the function would then return `boolean | undefined`. I'm not sure if converting to boolean is necessary but I did it in order to simplify the code in the client tests (I simply `expect(comp.canArchiveCourse()).toEqual(false);` instead of also checking against undefined.

Reply:
But I think when we use the new strict templates, we have to do these types of strict checks in general

Reply:
Ah yeah thats a smart move, didn't think of that 👍